src/gui/options_imp.cpp

```diff
@@ -403,6 +406,18 @@ void options_imp::saveOptions()
     Preferences::setTorrentFileAssoc(checkAssociateTorrents->isChecked());
     Preferences::setMagnetLinkAssoc(checkAssociateMagnetLinks->isChecked());
 #endif
+#ifdef Q_OS_MAC
+    if (checkAssociateTorrents->isChecked()) {
+        Preferences::setTorrentFileAssoc();
+        checkAssociateTorrents->setChecked(Preferences::isTorrentFileAssocSet());
+        checkAssociateTorrents->setEnabled(!checkAssociateTorrents->isChecked());
```

Comment:
>Saying simple, it is greyed out when we already have association.

I understand this. But why? Because we can't remove our association afterwards? (like @glassez says above?)